node/core/candidate-validation/src/lib.rs

```diff
@@ -269,70 +267,100 @@ async fn check_assumption_validation_data(
 	})
 }
 
-async fn spawn_validate_from_chain_state(
+async fn find_assumed_validation_data(
 	ctx: &mut impl SubsystemContext<Message = CandidateValidationMessage>,
-	execution_mode: ExecutionMode,
-	descriptor: CandidateDescriptor,
-	pov: Arc<PoV>,
-	spawn: impl SpawnNamed + 'static,
-) -> SubsystemResult<Result<ValidationResult, ValidationFailed>> {
+	descriptor: &CandidateDescriptor,
+) -> SubsystemResult<AssumptionCheckOutcome> {
 	// The candidate descriptor has a `persisted_validation_data_hash` which corresponds to
 	// one of up to two possible values that we can derive from the state of the
 	// relay-parent. We can fetch these values by getting the persisted validation data
 	// based on the different `OccupiedCoreAssumption`s.
-	match check_assumption_validation_data(
-		ctx,
-		&descriptor,
+
+	const ASSUMPTIONS: &[OccupiedCoreAssumption] = &[
 		OccupiedCoreAssumption::Included,
-	).await? {
-		AssumptionCheckOutcome::Matches(validation_data, validation_code) => {
-			return spawn_validate_exhaustive(
-				ctx,
-				execution_mode,
-				validation_data.persisted,
-				Some(validation_data.transient),
-				validation_code,
-				descriptor,
-				pov,
-				spawn,
-			).await;
-		}
-		AssumptionCheckOutcome::DoesNotMatch => {},
-		AssumptionCheckOutcome::BadRequest => return Ok(Err(ValidationFailed("Bad request".into()))),
+		OccupiedCoreAssumption::TimedOut,
+		// TODO: Why don't we check `Free`? The guide assumes there are only two possible assumptions.
```

Comment:
`Included` and `TimedOut` on a free core are both no-ops, so any candidate which assumes the core is free would already pass one of those. Adding the third check is a no-op.

`Free` is kind of an odd one out because there are very few situations where you actually want to check to see that a core is fully free. Off the top of my head, I can't think of any.